modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/reagents/incubus_draft.dm

```diff
@@ -0,0 +1,194 @@
+		var/target_bodypart = exposed_mob.get_bodypart(BODY_ZONE_PRECISE_GROIN)
+		if(prob(damage_chance))
+			to_chat(exposed_mob, span_danger("You feel a tightness in your pants!"))
+			exposed_mob.apply_damage(1, BRUTE, target_bodypart)
+
+	return ..()
+
+/datum/reagent/drug/aphrodisiac/incubus_draft/overdose_effects(mob/living/carbon/human/exposed_mob)
+	if(exposed_mob.client?.prefs.read_preference(/datum/preference/toggle/erp/penis_enlargement))
+		if(!exposed_mob.getorganslot(ORGAN_SLOT_PENIS) && exposed_mob.client?.prefs.read_preference(/datum/preference/toggle/erp/new_genitalia_growth))
+			var/list/data = species_to_penis[exposed_mob.dna.species.id]
+			if(!data)
+				data = species_to_penis[SPECIES_HUMAN]
+
+			exposed_mob.dna.features["penis_sheath"] = data["sheath"]
+			exposed_mob.dna.mutant_bodyparts[ORGAN_SLOT_PENIS][MUTANT_INDEX_NAME] = data["index_name"]
+			exposed_mob.dna.mutant_bodyparts[ORGAN_SLOT_TESTICLES][MUTANT_INDEX_NAME] = data["balls"]
+			var/colour = data["colour"]
+			if(colour)
+				exposed_mob.dna.mutant_bodyparts[ORGAN_SLOT_PENIS][MUTANT_INDEX_COLOR_LIST] = list(colour)
+
+			if(!exposed_mob.getorganslot(ORGAN_SLOT_TESTICLES))
+				var/obj/item/organ/balls_path = /obj/item/organ/external/genital/testicles
+				balls_path = new /obj/item/organ/external/genital/testicles
+				balls_path.build_from_dna(exposed_mob.dna, ORGAN_SLOT_TESTICLES)
+				balls_path.Insert(exposed_mob, 0, FALSE)
+				var/obj/item/organ/external/genital/new_balls = exposed_mob.getorganslot(ORGAN_SLOT_TESTICLES)
+				new_balls.genital_size = 1
+				new_balls.update_sprite_suffix()
+
+			var/obj/item/organ/external/genital/penis/new_penis = new /obj/item/organ/external/genital/penis
+			new_penis.build_from_dna(exposed_mob.dna, ORGAN_SLOT_PENIS)
+			new_penis.Insert(exposed_mob, 0, FALSE)
+			new_penis.genital_size = 4
+			new_penis.girth = 3
+			new_penis.update_sprite_suffix()
+			exposed_mob.update_body()
+			to_chat(exposed_mob, span_purple("Your crotch feels warm as something suddenly sprouts between your legs."))
+		// Makes the balls bigger if they're small.
+		var/obj/item/organ/external/genital/testicles/mob_testicles = exposed_mob.getorganslot(ORGAN_SLOT_TESTICLES)
+		if(mob_testicles)
+			if(mob_testicles.genital_size > 2)
+				return
+			mob_testicles.genital_size = 2
+	// Separates gender change stuff from cock growth.
+	var/obj/item/organ/external/genital/breasts/mob_breasts = exposed_mob.getorganslot(ORGAN_SLOT_BREASTS)
+	if(exposed_mob.client?.prefs.read_preference(/datum/preference/toggle/erp/gender_change))
+		if(exposed_mob.gender == FEMALE)
+			exposed_mob.set_gender(MALE)
+			exposed_mob.physique = exposed_mob.gender
+			exposed_mob.update_body()
+			exposed_mob.update_mutations_overlay()
+		if(!mob_breasts)
+			return
+		if(exposed_mob.client?.prefs.read_preference(/datum/preference/toggle/erp/breast_enlargement)) // To do breast shrinkage, check if prefs allow for this.
```

Comment:
There should be question marks following any question mark, because if one can be null, the ones afterwards will be affected too.